vendor/build.gradle

```diff
@@ -0,0 +1,168 @@
+project(':vendor') {
+  apply plugin: 'cpp'
+
+  model {
+    components {
+      // We are building protoc and protobuf from source for maximum compatibility
+      // across platform. Please note that we are using a very minimilistic
+      // port of the build to Gradle.
```

Comment:
Did you write these build configurations by hand? So, are you not using the build scripts that come with protobuf at all? (which is totally fine. I am just curious)